tests/components/nightscout/test_sensor.py

```diff
@@ -0,0 +1,60 @@
+"""The sensor tests for the Nightscout platform."""
+
+from homeassistant.components.nightscout.const import (
+    ATTR_DATE,
+    ATTR_DELTA,
+    ATTR_DEVICE,
+    ATTR_DIRECTION,
+    ATTR_SVG,
+)
+from homeassistant.const import ATTR_ICON, STATE_UNAVAILABLE
+
+from tests.components.nightscout import (
+    GLUCOSE_READINGS,
+    init_integration,
+    init_integration_empty_response,
+    init_integration_unavailable,
+)
+
+
+async def test_sensor_state(hass):
+    """Test sensor state data."""
+    await init_integration(hass)
+
+    test_glucose_sensor = hass.states.get("sensor.blood_sugar")
+    assert test_glucose_sensor.state == str(
+        GLUCOSE_READINGS[0].sgv  # pylint: disable=maybe-no-member
+    )
+
+
+async def test_sensor_error(hass):
+    """Test sensor state data."""
+    await init_integration_unavailable(hass)
+
+    test_glucose_sensor = hass.states.get("sensor.blood_sugar")
+    assert test_glucose_sensor.state == STATE_UNAVAILABLE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)